src/platforms/alexa/apis/InSkillPurchase.ts

```diff
@@ -0,0 +1,96 @@
+import { RequestEnvelope } from "ask-sdk-model";
+import * as _ from "lodash";
+import * as rp from "request-promise";
+
+import { ConnectionsSendRequest } from "../directives";
+
+export class InSkillPurchase {
+  public static buy(productId: string, token: string) {
+    const payload: any = {
+      InSkillProduct: {
+        productId,
+      },
+    };
+
+    return new ConnectionsSendRequest("Buy", payload, token);
+  }
+
+  public static cancel(productId: string, token: string) {
+    const payload: any = {
+      InSkillProduct: {
+        productId,
+      },
+    };
+
+    return new ConnectionsSendRequest("Cancel", payload, token);
+  }
+
+  public static upsell(productId: string, upsellMessage: string, token: string) {
+    const payload: any = {
+      InSkillProduct: {
+        productId,
+      },
+      upsellMessage,
+    };
+
+    return new ConnectionsSendRequest("Upsell", payload, token);
+  }
+
+  public voxaEvent: any; // the event as sent by the service
+
+  constructor(event: RequestEnvelope) {
+    this.voxaEvent = _.cloneDeep(event);
```

Comment:
check this, you're passing a `voxaEvent` as any in the calling `AlexaEvent` class, however this class takes a `RequestEnvelope` in the constructor